packages/cmf/README.md

```diff
@@ -36,62 +36,113 @@ It fits with our goal, this is why this add-on has been named that way.
+The action may change the application state, which in turn may change some components in the user interface.
 
 ## Definitions
 
-We have the following objects to build a user interface:
+### Components
 
-* props
-* actions
+An HTML page is composed by a tree structure called the DOM. In our case React manage that DOM
+using a tree of `components`.
 
-Let's talk about each of them.
+`Components` in the context of CMF are React components.
 
-### Props
+But they are not just React component. We give them some super power using [cmfConnect](./src/cmfConnect.md).
 
-Props are special React components. 
-They are high level component which has the following responsibility:
-They must dispatch props to configurable components.
+```javascript
+import React from 'react';
+import { cmfConnect } from '@talend/react-cmf';
+
+function MyComponent(props) {
+	return (
+		<article>
+			<h2>{props.title}</h2>
+			<button onClick={() => props.dispatchActionCreator('read')}>Read</button>
+		</article>
+	);
+}
+export default cmfConnect({})(MyComponent);
+```
 
-They are called by UI abstraction library from the router and connected to the store throw the settings.
+### Actions
 
-So a view is can be a pure component.
+Actions are [redux actions](http://redux.js.org/docs/basics/Actions.html).
 
-Then view will be composed of react components that can get their props.
+## Store structure
 
-### Actions
+cmf store structure is the following
 
-Actions are [redux actions](http://redux.js.org/docs/basics/Actions.html).
+* root
+  * cmf
+    * collections
+    * components
+    * settings
 
 ### ComponentState Management
 
 Component state can be easily stored in cmf state, each are identified by their name and an unique key,
-so component state can be stored and reused later
+so component state can be stored and reused later.
+
+We give you the choice to use either:
+
+* CMF redux state (this.props.state && this.props.setState)
+* React component state (this.state && this.setState)
+
+Warning: you should use the redux state except for part that require lots of mutation without sharing.
+For example for Forms you should prefer to use the internal React component state.
+
 
 ### Collections management
 
-Manage a local cache of your business data
+Manage a local cache of your business data.
+You can connect your component to give him access to your data and being able
+to dispatch action to let CMF's reducers write them.
 
-## Internals: The registry
+You can dispatch some actionCreator in `api.actions.collections` for that
 
-You will find the registry as the central piece of ui abstraction.
-It's just a key/object registry and it's used with prefix to store the following:
+## Configuration (settings)
 
-* action creators (function)
-* views (React Component)
+We don't want to have to rebuild our app to change a label of a button right ?
```

Comment:
We don't want to rewrite a bunch of app code to change a label of a button right ?